packages/frontend/app/models/identity.ts

```diff
@@ -22,6 +22,10 @@ export default class Identity extends Model implements Partial<PublicKey> {
     return toHex(this.publicKey);
   }
 
+  get publicSigningKeyAsHex() {
+    return toHex(this.publicSigningKey);
```

Comment:
what happens when someone doesn't have a publicSigningKey?

Reply:
I believe based on the work in PR #690, every user should have a public signing key as part of their user, regardless of whether they are new or returning users.

Reply:
Users before that PR wouldn't have signing keys ;)
(I ran in to this myself with my own login, and that's why I added the migrate method to `load`) :P

Reply:
With settings or mnemonic?

Reply:
both

Reply:
Doesn't the new login worker generate new signing keys upon login with mnemonic? Let me look into settings login a little more.

Reply:
For settings, we would need to add an else statement to importFromKey here:https://github.com/NullVoxPopuli/emberclear/blob/master/packages/frontend/app/services/current-user.ts such that when a settings file doesn't have a signing key, it generates a new one upon login.

Reply:
> Doesn't the new login worker generate new signing keys upon login with mnemonic? Let me look into settings login a little more.

you right. if you login with your mnemonic, then you get a new signing key.

so it's just the settings upload that didn't mirror that behavior. 
anywho, it's resolved now :)

Reply:
Looks like I may have misremembered again! sorry.

This is the line I added: https://github.com/NullVoxPopuli/emberclear/blob/master/packages/frontend/app/services/current-user.ts#L137

that accounts for if you are already logged in, and your browser suddenly has the code that uses signing keys

Reply:
Added code for this in #839

Reply:
Ah, if you stay logged in, we would need to generate.

Reply:
I believe we would just need to add checks in the getters for public and private signing keys and generate if either do not exist. Right now if you get the record using currentUser(), like I believe all the code should, it will call that migrate and generate them if they don't exist. Only in the case where those specific getters are called will there be the possibility that they won't exist. That is if none of the code tries to access record directly.

Reply:
Should I add that to #839 @NullVoxPopuli ?

Reply:
no need for right now -- keep focussed on the channel stuff ;)